Algorithms/Numeric/AutomorphicNumber.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+
+namespace Algorithms.Numeric
+{
+    /// <summary>
+    /// Calculates Automorphic numbers. A number is said to be an Automorphic number
+    /// if the square of the number will contain the number itself at the end.
+    /// </summary>
+    public static class AutomorphicNumber
+    {
+        /// <summary>
+        /// Generates a list of automorphic numbers that are between <paramref name="lowerBound"/> and <paramref name="upperBound"/>
+        /// inclusive.
+        /// </summary>
+        /// <param name="lowerBound">The lower bound of the list.</param>
+        /// <param name="upperBound">The upper bound of the list.</param>
+        /// <returns>A list that contains all of the automorphic numbers between <paramref name="lowerBound"/> and <paramref name="upperBound"/> inclusive.</returns>
+        /// <exception cref="ArgumentException">If the <paramref name="lowerBound"/>
+        /// or <paramref name="upperBound"/> is not greater than zero
+        /// or <paramref name="upperBound"/>is lower than the <paramref name="lowerBound"/>.</exception>
+        public static IEnumerable<long> GetAutomorphicNumbers(long lowerBound, long upperBound)
+        {
+            if (lowerBound < 1)
+            {
+                throw new ArgumentException($"Lower Bound must be greater than 0: Actual value {lowerBound}");
+            }
+
+            if (upperBound < 1)
+            {
+                throw new ArgumentException($"Upper Bound must be greater than 0: Actual value {upperBound}");
+            }
+
+            if (lowerBound > upperBound)
+            {
+                throw new ArgumentException($"Lower Bound must be lower than Upper Bound: Actual values [{lowerBound},{upperBound}]");
+            }
```

Comment:
Based on the check you make, the message must say that 'The lower bound must be less than or equal to the upper bound'

Reply:
No need to add actual value to exception message, the caller knows what he passed to the method

Reply:
In this setting yes, but it would be better to output the actual values when for example this algorithm is used in a system that logs exceptions to a log file.

Reply:
Or it displays errors to the end user, and you don't want them to see the values used in the code. So it depends, but adding data higher in the callstack is easy, while removing it is tedious.